src/components/hooks/use-livechat.js

```diff
@@ -0,0 +1,145 @@
+        `
+        document.body.appendChild(livechat_script)
+        if (callback) callback()
+    }
+
+    React.useEffect(() => {
+        let cookie_interval = null
+        let script_timeout = null
+        if (isBrowser()) {
+            const domain = window.location.hostname.includes('deriv.com')
+                ? 'deriv.com'
+                : 'binary.sx'
+            import('@livechat/customer-sdk').then((CSDK) => {
+                CustomerSdk.current = CSDK
+            })
+            const checkCookie = (() => {
+                let lastCookie = document.cookie // 'static' memory between function calls
+                return function () {
+                    const currentCookie = document.cookie
+                    if (currentCookie != lastCookie) {
+                        const client_information = Cookies.get('client_information', {
+                            domain,
+                        })
+                        setLoggedIn(!!client_information)
+                        lastCookie = currentCookie // store latest cookie
+                    }
+                }
+            })()
+
+            cookie_interval = setInterval(checkCookie, 500)
+
+            // The purpose is to load the script after everything is load but not async or defer. Therefore, it will be ignored in the rendering timeline
+            script_timeout = setTimeout(() => {
+                loadLiveChatScript(() => {
+                    window.LiveChatWidget.on('ready', () => {
+                        setLiveChatInteractive(true)
+                        if (is_livechat_query?.toLowerCase() === 'true') {
+                            window.LC_API.open_chat_window()
+                        }
+                    })
+                })
+            }, 2000)
+        }
+
+        return () => {
+            clearInterval(cookie_interval)
+            clearTimeout(script_timeout)
+        }
+    }, [])
+
+    React.useEffect(() => {
+        if (isBrowser()) {
+            let customerSDK = null
+            const domain = window.location.hostname.includes('deriv.com')
+                ? 'deriv.com'
+                : 'binary.sx'
+            if (CustomerSdk.current) {
+                customerSDK = CustomerSdk.current.init({
+                    licenseId: livechat_license_id,
+                    clientId: livechat_client_id,
+                })
+            }
+            if (is_livechat_interactive) {
+                window.LiveChatWidget.on('ready', () => {
+                    let session_variables = {
+                        loginid: '',
+                        landing_company_shortcode: '',
+                        currency: '',
+                        residence: '',
+                        email: '',
+                    }
```

Comment:
can we omit this declaration and declare it [here](https://github.com/binary-com/deriv-com/pull/1351/files#diff-e5e90c60d24e7dae6576024383c266127f2d7aef9cb1717329e7e557cc98fe9fR105) instead with fallback of empty string